pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -498,6 +503,38 @@ func (a *Actuator) ensureAnnotation(ctx context.Context, machine *machinev1beta1
 	return true, a.client.Update(ctx, machine)
 }
 
+func (a *Actuator) ensureProviderID(ctx context.Context, machine *machinev1beta1.Machine, host *bmh.BareMetalHost) (error) {
+	providerID := "baremetalhost:///" + host.Namespace + "/" + host.Name
+	existingProviderID := machine.Spec.ProviderID
+	if existingProviderID == nil || *existingProviderID != providerID {
+		log.Printf("Setting ProviderID %s for machine %s.", providerID, machine.Name)
+		machine.Spec.ProviderID = &providerID
+		err := a.client.Update(ctx, machine)
+		if err != nil {
+			log.Printf("Failed to update machine ProviderID, error: %s", err.Error())
+			return err
+		}
+	}
+
+	node, err := a.getNodeByMachine(ctx, machine)
+        if err != nil {
+                log.Printf("Failed to find node assosicated with Machine %s, error: %s", machine.Name, err.Error())
+//                return err
+		// FIXME should we ignore this?
+		return nil
```

Comment:
@dhellmann @mhrivnak pls can you check this updated version

I'm only calling ensureProviderID in the `Update()` atm (as we know the NodeRef won't be set at the time of `Create()`) - but even in `Update()` it's possible we have some update prior to the NodeRef getting set, so what's the correct way to handle that?

I can return the err but then I guess we'll log errors for an expected/temporary event (and it breaks all the tests but I can fix that :) ) - for now I'm returning nil to enable further testing but I'd like your feedback before I go fixing tests etc, thanks!

Reply:
As long as the system is eventually consistent, I don't think this is an error. It's expected even. If you really want to log something, I could see an INFO-level message such as `Node does not yet exist for Machine %s`.